h2/src/main/org/h2/mvstore/Page.java

```diff
@@ -709,10 +716,18 @@ protected final int write(Chunk chunk, WriteBuffer buff, List<Long> toc) {
                     compressor = store.getCompressorHigh();
                     compressType = DataUtils.PAGE_COMPRESSED_HIGH;
                 }
-                byte[] exp = new byte[expLen];
-                buff.position(compressStart).get(exp);
                 byte[] comp = new byte[expLen * 2];
-                int compLen = compressor.compress(exp, expLen, comp, 0);
+                ByteBuffer byteBuffer = buff.getBuffer();
+                int pos = 0;
+                byte[] exp;
+                if (byteBuffer.hasArray()) {
+                    exp = byteBuffer.array();
+                    pos = compressStart;
```

Comment:
Here we need `pos = byteBuffer.arrayOffset() + compressStart` I guess.

Reply:
Indeed, it's currently always zero, added